### PR TITLE
More HDF5 checks

### PR DESCRIPTION
**Pull request by masterleinad**
`coverity` noticed that the return type for some of the `HDF5` functions were not checked. This pull request fixes that.

### CODE REVIEW

File: source/base/hdf5.cc [hunk 1]
```diff
@@ -588,13 +601,17 @@ namespace HDF5
     , global_no_collective_cause(H5D_MPIO_SET_INDEPENDENT)
   {
     hdf5_reference = std::shared_ptr<hid_t>(new hid_t, [](hid_t *pointer) {
-      // Relase the HDF5 resource
-      H5Dclose(*pointer);
+      // Release the HDF5 resource
+      const herr_t ret = H5Dclose(*pointer);
+      Assert(ret >= 0, ExcInternalError());
+      (void)ret;
```
Comment (drwells) [inline source/base/hdf5.cc]: I believe these are custom deleters, so we should use `AssertNothrow` here and below.

File: source/base/hdf5.cc [hunk 2]
```diff
@@ -1419,7 +1444,7 @@ namespace HDF5
 
     if (mpi)
       {
-        // Relase the HDF5 resource
+        // Release the HDF5 resource
```
Comment (drwells) [inline source/base/hdf5.cc]: :)

File: source/base/hdf5.cc (hunk 1 above)
Comment (masterleinad) [reply source/base/hdf5.cc]: Good point! I fixed all the places I could find in that file.

**Comment by masterleinad** [comment]
It seems that the status of `ready` is not forwarded. https://jenkins.tjhei.info/blue/organizations/jenkins/dealii-mark/detail/PR-7736/3/pipeline shows that it is passing.

**Comment by drwells** [comment]
Since all of the testers passed lets merge.

**Pull request merged**